### PR TITLE
Feature/momentum bias

### LINKED ISSUE #312: Add capability to desaturate reaction wheels to momentum bias
**Describe your use case**
The current implementation of the momentum management module only lowers the magnitude of the reaction wheel momentum. It would be useful to bias the reaction wheel momenta in preparation for events that will load up the wheels in a specified direction.

**Describe alternatives solutions you've considered**
The current version works well to lower the total momentum, but can't bias it in a specified direction.

**Additional context**
N/A.

### PR DESCRIPTION
**Pull request by joaogvcarneiro**
* **Tickets addressed:** #312 
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description

The first three commits change the source files, the next two add two new unit tests, and the last one improves the module's documentation.

## Verification

Two tests are added. The first one checks the instantaneous value of the momentum management torque to check it is applied from the current to the desired rw momentum. The second is an integrated test that runs through the entire momentum dumping routine and checks that the last value of the rw momentum is close to its desired value.

## Documentation

Added documentation that explains this new use case.

## Future work

N/A.

### CODE REVIEW

File: src/fswAlgorithms/attControl/thrMomentumManagementCpp/_UnitTest/test_thrMomentumManagementCpp.py
Comment (patkenneally) [inline src/fswAlgorithms/attControl/thrMomentumManagementCpp/_UnitTest/test_thrMomentumManagementCpp.py]: :small_red_triangle_down: With one value there's no need to parameterize this test. We can set the value directly from within the test.

File: src/fswAlgorithms/attControl/thrMomentumManagementCpp/_UnitTest/test_thrMomentumManagementCpp.py
Comment (patkenneally) [inline src/fswAlgorithms/attControl/thrMomentumManagementCpp/_UnitTest/test_thrMomentumManagementCpp.py]: :small_red_triangle_down: You can just set this explicitly in the assert call.

File: src/fswAlgorithms/attControl/thrMomentumManagementCpp/_UnitTest/test_thrMomentumManagementCpp_integrated.py
Comment (patkenneally) [inline src/fswAlgorithms/attControl/thrMomentumManagementCpp/_UnitTest/test_thrMomentumManagementCpp_integrated.py]: :small_red_triangle_down: One value, so doesn't need to be parameterized.

**Review by patkenneally — CHANGES_REQUESTED** [review]
Legend:
:small_red_triangle_down: *Issues* to address before merge
:large_orange_diamond: *Requests* that should not block merge, but should at least be discussed
:large_blue_circle: *Recommendations* that can be ignored if desired

I'm still looking but here are the things that jumped out at me.

File: src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.h
Comment (patkenneally) [inline src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.h]: :large_orange_diamond: Including Eignen/Core was probably better due to it's narrower scope. Was there a reason to include the broader Eigen/Dense?

File: src/fswAlgorithms/attControl/thrMomentumManagementCpp/_UnitTest/test_thrMomentumManagementCpp.py
Comment (joaogvcarneiro) [reply src/fswAlgorithms/attControl/thrMomentumManagementCpp/_UnitTest/test_thrMomentumManagementCpp.py]: Done.

File: src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.h
Comment (joaogvcarneiro) [reply src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.h]: Not really, I'll switch back to Eigen/Core.

File: src/fswAlgorithms/attControl/thrMomentumManagementCpp/_UnitTest/test_thrMomentumManagementCpp_integrated.py
Comment (joaogvcarneiro) [reply src/fswAlgorithms/attControl/thrMomentumManagementCpp/_UnitTest/test_thrMomentumManagementCpp_integrated.py]: Done.

File: src/fswAlgorithms/attControl/thrMomentumManagementCpp/_UnitTest/test_thrMomentumManagementCpp.py
Comment (joaogvcarneiro) [reply src/fswAlgorithms/attControl/thrMomentumManagementCpp/_UnitTest/test_thrMomentumManagementCpp.py]: Done.

Comment (patkenneally) [reply src/fswAlgorithms/attControl/thrMomentumManagementCpp/_UnitTest/test_thrMomentumManagementCpp.py]: No quite. Need to remove`hd_B` from the function prototype.

File: src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.h
Comment (patkenneally) [reply src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.h]: While you don't have to adjust, the best way to have reverted his change would have been to rebase and edit the commit to remove the change all together.

File: src/fswAlgorithms/attControl/thrMomentumManagementCpp/_UnitTest/test_thrMomentumManagementCpp.py
Comment (joaogvcarneiro) [reply src/fswAlgorithms/attControl/thrMomentumManagementCpp/_UnitTest/test_thrMomentumManagementCpp.py]: Ah, I took it out of one but not the other. It should be fixed now.

**Review by patkenneally — APPROVED** [review]

**Pull request merged**
